### PR TITLE
add OLMv1 cap

### PR DESCRIPTION
**Pull request by jianzhangbjz**
Address https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/logs/periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-gcp-ipi-basecap-none-additionalcaps-amd-f28-destructive/1857067594461220864 
```console
===check .status.capabilities.knownCapabilities
ERROR: known capabilities does not match with cvo .status.capabilities.knownCapabilities!
cvo_caps: Build CSISnapshot CloudControllerManager CloudCredential Console DeploymentConfig ImageRegistry Ingress Insights MachineAPI NodeTuning OperatorLifecycleManager OperatorLifecycleManagerV1 Storage baremetal marketplace openshift-samples
known capability set: Build CSISnapshot CloudControllerManager CloudCredential Console DeploymentConfig ImageRegistry Ingress Insights MachineAPI NodeTuning OperatorLifecycleManager Storage baremetal marketplace openshift-samples
```

### CODE REVIEW

**Comment by jianzhangbjz** [comment]
Hi @jinyunma , could you help have a review? Thanks!

**Comment by jianzhangbjz** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-gcp-ipi-basecap-none-additionalcaps-amd-f28-destructive

**Comment by jinyunma** [comment]
Following lines also need to add new cap.
1. https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/conf/capability/baselinecaps/vset/ipi-conf-capability-baselinecaps-vset-commands.sh#L71
2. https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/conf/capability/additionalcaps/ipi-conf-capability-additionalcaps-commands.sh#L68

step [cucushift-upgrade-check-capability](https://github.com/openshift/release/blob/master/ci-operator/step-registry/cucushift/upgrade/check/capability/cucushift-upgrade-check-capability-commands.sh) should have similar update as `cucushift-installer-check-capability`

cc @jianlinliu @jiajliu

File: ci-operator/step-registry/ipi/conf/manual-creds/remove-unnecessary-creds/ipi-conf-manual-creds-remove-unnecessary-creds-commands.sh [hunk 1]
```diff
@@ -61,7 +61,7 @@ v414=" ${v413} MachineAPI Build DeploymentConfig ImageRegistry"
 v415=" ${v414} OperatorLifecycleManager CloudCredential"
 v416=" ${v415} CloudControllerManager Ingress"
 v417=" ${v416}"
-v418=" ${v417}"
+v418=" ${v417} OperatorLifecycleManagerV1"
```
Comment (jianlinliu) [inline ci-operator/step-registry/ipi/conf/manual-creds/remove-unnecessary-creds/ipi-conf-manual-creds-remove-unnecessary-creds-commands.sh:64]: Beside these two step need to be updated, the following step also need to be updated:
```
ipi-conf-capability-additionalcaps
ipi-conf-capability-baselinecaps-vset
cucushift-upgrade-check-capability
```

**Comment by jianzhangbjz** [comment]
> Following lines also need to add new cap.
> https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/conf/capability/baselinecaps/vset/ipi-conf-capability-baselinecaps-vset-commands.sh#L71
https://github.com/openshift/release/blob/master/ci-operator/step-registry/ipi/conf/capability/additionalcaps/ipi-conf-capability-additionalcaps-commands.sh#L68
Added them.

> step [cucushift-upgrade-check-capability](https://github.com/openshift/release/blob/master/ci-operator/step-registry/cucushift/upgrade/check/capability/cucushift-upgrade-check-capability-commands.sh) should have similar update as cucushift-installer-check-capability

I didn't find the related config for 4.17 and 4.18, not sure if any specific configs, ignored them in this PR.

File: ci-operator/step-registry/ipi/conf/manual-creds/remove-unnecessary-creds/ipi-conf-manual-creds-remove-unnecessary-creds-commands.sh (hunk 1 above)
Comment (jianzhangbjz) [reply ci-operator/step-registry/ipi/conf/manual-creds/remove-unnecessary-creds/ipi-conf-manual-creds-remove-unnecessary-creds-commands.sh:64]: > ipi-conf-capability-additionalcaps
> ipi-conf-capability-baselinecaps-vset

Added them.
> cucushift-upgrade-check-capability

I didn't find the related config for 4.17 and 4.18 there, not sure if any specific configs, ignored them in this PR, left it to the step owner.

**Comment by jianzhangbjz** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-gcp-ipi-basecap-none-additionalcaps-amd-f28-destructive

**Comment by jinyunma** [comment]
job `periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-gcp-ipi-basecap-none-additionalcaps-amd-f28-destructive` failed because installer does not bump the latest API version and does not know new cap `OperatorLifecycleManagerV1`, need to waiting for https://github.com/openshift/installer/pull/9196

**Comment by jianzhangbjz** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-gcp-ipi-basecap-none-additionalcaps-amd-f28-destructive

**Comment by jianzhangbjz** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-azure-ipi-basecap-vset-arm-f2

**Comment by jianlinliu** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-4.18-upgrade-from-stable-4.17-aws-ipi-sno-etcd-encryption-basecap-none-arm-f28

**Comment by jianzhangbjz** [comment]
https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/58914/rehearse-58914-periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-4.18-upgrade-from-stable-4.17-aws-ipi-sno-etcd-encryption-basecap-none-arm-f28/1858789614408437760/artifacts/aws-ipi-sno-etcd-encryption-basecap-none-arm-f28/cucushift-upgrade-check-capability/build-log.txt failed at:
```console
------check enabled capabilities-----
enabled capability set: CloudControllerManager CloudCredential Ingress MachineAPI OperatorLifecycleManagerV1
check capability CloudControllerManager
check capability CloudCredential
check capability Ingress
check capability MachineAPI
check capability OperatorLifecycleManagerV1
ERROR: capability OperatorLifecycleManagerV1: operator olm -- not found!
```
That's because the OLMv1 still in TP, waiting for https://github.com/openshift/cluster-olm-operator/pull/74 merged.

**Comment by jianlinliu** [comment]
> That's because the OLMv1 still in TP, waiting for [openshift/cluster-olm-operator#74](https://github.com/openshift/cluster-olm-operator/pull/74) merged.

Do you mean  OLMv1 is introduced and available in 4.18 ? and in 4.18, it is an optional operator ?

**Comment by jianzhangbjz** [comment]
> Do you mean OLMv1 is introduced and available in 4.18 ? and in 4.18, it is an optional operator ?

For now, the OLMV1 is TP, and will be GA in 4.18.

File: ci-operator/step-registry/cucushift/installer/check/capability/cucushift-installer-check-capability-commands.sh [hunk 2]
```diff
@@ -74,6 +74,7 @@ caps_operator[NodeTuning]="node-tuning"
 caps_operator[MachineAPI]="machine-api control-plane-machine-set cluster-autoscaler"
 caps_operator[ImageRegistry]="image-registry"
 caps_operator[OperatorLifecycleManager]="operator-lifecycle-manager operator-lifecycle-manager-catalog operator-lifecycle-manager-packageserver"
+caps_operator[OperatorLifecycleManagerV1]="olm"
```
Comment (evakhoni) [inline ci-operator/step-registry/cucushift/installer/check/capability/cucushift-installer-check-capability-commands.sh:77]: so its caps_operator, not caps_resource? I don't see "olm" mentioned in `oc get co`, however `oc get olm` works..

**Comment by evakhoni** [comment]
> https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/58914/rehearse-58914-periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-4.18-upgrade-from-stable-4.17-aws-ipi-sno-etcd-encryption-basecap-none-arm-f28/1858789614408437760/artifacts/aws-ipi-sno-etcd-encryption-basecap-none-arm-f28/cucushift-upgrade-check-capability/build-log.txt failed at:
> 
> ```
> ------check enabled capabilities-----
> enabled capability set: CloudControllerManager CloudCredential Ingress MachineAPI OperatorLifecycleManagerV1
> check capability CloudControllerManager
> check capability CloudCredential
> check capability Ingress
> check capability MachineAPI
> check capability OperatorLifecycleManagerV1
> ERROR: capability OperatorLifecycleManagerV1: operator olm -- not found!
> ```
> 
> That's because the OLMv1 still in TP, waiting for [openshift/cluster-olm-operator#74](https://github.com/openshift/cluster-olm-operator/pull/74) merged.

hmm.. I think the PRs already merged into 4.18 ec4. is it supposed to be a standard operator, visible in oc get co? tried on that cluster and I don't see any.. but `oc get olm` returns a resource.. is this resource what we're looking for?

**Comment by jianzhangbjz** [comment]
>hmm.. I think the PRs already merged into 4.18 ec4. 

I guess not. OLMv1 doesn't GA in 4.18 ec4, but in the latest accepted nightly payload.
```console
jiazha-mac:~ jiazha$ oc adm release info -a .dockerconfigjson --commits quay.io/openshift-release-dev/ocp-release:4.18.0-ec.4-x86_64 |grep olm 
  cluster-olm-operator                           https://github.com/openshift/cluster-olm-operator                           a88e5a40bb2bb53ff87d3983f9555430b188daaa
  olm-catalogd                                   https://github.com/openshift/operator-framework-catalogd                    2f39d4891541553f22b697ba49baf1ebb4ea4e72
  olm-operator-controller                        https://github.com/openshift/operator-framework-operator-controller         f8d2b64065e38d4c0e3adfceb6b2c1a09834fc20

jiazha-mac:~ jiazha$ oc adm release info -a .dockerconfigjson --commits registry.ci.openshift.org/ocp/release:4.18.0-0.nightly-2024-11-22-231049 |grep olm 
  cluster-olm-operator                           https://github.com/openshift/cluster-olm-operator                           67427dd38d38af1c09c70292676f7b80936cce3e
  olm-catalogd                                   https://github.com/openshift/operator-framework-catalogd                    bd22125eb5c737d7ce6764fc660a99e9ac85e1e6
  olm-operator-controller                        https://github.com/openshift/operator-framework-operator-controller         5c9e433bf69ae09b5269956ed8249b527dc8278d
```

> is it supposed to be a standard operator, visible in oc get co?

Yes.

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-gcp-ipi-basecap-none-additionalcaps-amd-f28-destructive

**Comment by jianzhangbjz** [comment]
The ARM platform uses the multi-arch payload, details: [aws-ipi-sno-etcd-encryption-basecap-none-arm-f28/cucushift-upgrade-check-capability/](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/58914/rehearse-58914-periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-4.18-upgrade-from-stable-4.17-aws-ipi-sno-etcd-encryption-basecap-none-arm-f28/1860855242157985792/artifacts/aws-ipi-sno-etcd-encryption-basecap-none-arm-f28/cucushift-upgrade-check-capability/build-log.txt) 
```console
openshift-controller-manager    4.18.0-0.nightly-multi-2024-11-22-231004   True        False         False      45m     
service-ca                      4.18.0-0.nightly-multi-2024-11-22-231004   True        False         False      141m    
------check enabled capabilities-----
...
check capability OperatorLifecycleManagerV1
ERROR: capability OperatorLifecycleManagerV1: operator olm -- not found!
```
However, this latest [multi-arch payload](https://multi.ocp.releases.ci.openshift.org/releasestream/4.18.0-0.nightly-multi/release/4.18.0-0.nightly-multi-2024-11-22-231004 ) hasn't included the OLMv1. So, the failure as expected.
```console
jiazha-mac:~ jiazha$ oc adm release info -a .dockerconfigjson --commits quay.io/openshift-release-dev/ocp-release-nightly@sha256:e587424c309bde30865d8623949cd2d81fbb19e4aa07fb533143dfc2ba14d094 |grep olm 
  cluster-olm-operator                           https://github.com/openshift/cluster-olm-operator                           67427dd38d38af1c09c70292676f7b80936cce3e
  olm-catalogd                                   https://github.com/openshift/operator-framework-catalogd                    bd22125eb5c737d7ce6764fc660a99e9ac85e1e6
  olm-operator-controller                        https://github.com/openshift/operator-framework-operator-controller         5c9e433bf69ae09b5269956ed8249b527dc8278d
```

**Comment by jianlinliu** [comment]
> this latest [multi-arch payload](https://multi.ocp.releases.ci.openshift.org/releasestream/4.18.0-0.nightly-multi/release/4.18.0-0.nightly-multi-2024-11-22-231004) hasn't included the OLMv1

Let us wait for a while, I guess OLMv1 will land mult-arch payload soon.

**Comment by jianlinliu** [comment]
> However, this latest [multi-arch payload](https://multi.ocp.releases.ci.openshift.org/releasestream/4.18.0-0.nightly-multi/release/4.18.0-0.nightly-multi-2024-11-22-231004) hasn't included the OLMv1. So, the failure as expected

The commit - `67427dd38d38af1c09c70292676f7b80936cce3e` is the latest commit in `cluster-olm-operator` repo, so sounds like it already includes https://github.com/openshift/cluster-olm-operator/pull/74, right?

**Comment by jianzhangbjz** [comment]
Trigger one without the muti-arch payload.
/pj-rehearse periodic-ci-openshift-openshift-tests-private-release-4.18-amd64-nightly-aws-ipi-disc-priv-sts-basecap-none-f14

**Comment by jinyunma** [comment]
@jianlinliu @jianzhangbjz from last failed job [cvo log](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/58914/rehearse-58914-periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-4.18-upgrade-from-stable-4.17-aws-ipi-sno-etcd-encryption-basecap-none-arm-f28/1860855242157985792/artifacts/aws-ipi-sno-etcd-encryption-basecap-none-arm-f28/gather-extra/artifacts/pods/openshift-cluster-version_cluster-version-operator-7bd84cd8f7-drzlx_cluster-version-operator.log), seems that olm operator creation is skipped. Is it expected?
```
I1125 04:08:30.079162       1 sync_worker.go:1019] Skipping precreation of clusteroperator "olm" (722 of 902): disabled capabilities: OperatorLifecycleManagerV1
I1125 04:08:30.079179       1 sync_worker.go:1036] Running sync for olm "cluster" (715 of 902)
I1125 04:08:30.079189       1 sync_worker.go:1039] Skipping olm "cluster" (715 of 902): disabled capabilities: OperatorLifecycleManagerV1
I1125 04:08:30.079197       1 sync_worker.go:1036] Running sync for namespace "openshift-cluster-olm-operator" (716 of 902)
I1125 04:08:30.079208       1 sync_worker.go:1039] Skipping namespace "openshift-cluster-olm-operator" (716 of 902): disabled capabilities: OperatorLifecycleManagerV1
I1125 04:08:30.079215       1 sync_worker.go:1036] Running sync for clusterrole "cluster-olm-operator" (717 of 902)
I1125 04:08:30.079225       1 sync_worker.go:1039] Skipping clusterrole "cluster-olm-operator" (717 of 902): disabled capabilities: OperatorLifecycleManagerV1
I1125 04:08:30.079233       1 sync_worker.go:1036] Running sync for serviceaccount "openshift-cluster-olm-operator/cluster-olm-operator" (718 of 902)
I1125 04:08:30.079250       1 sync_worker.go:1039] Skipping serviceaccount "openshift-cluster-olm-operator/cluster-olm-operator" (718 of 902): disabled capabilities: OperatorLifecycleManagerV1
I1125 04:08:30.079261       1 sync_worker.go:1036] Running sync for service "openshift-cluster-olm-operator/cluster-olm-operator-metrics" (719 of 902)
I1125 04:08:30.079271       1 sync_worker.go:1039] Skipping service "openshift-cluster-olm-operator/cluster-olm-operator-metrics" (719 of 902): disabled capabilities: OperatorLifecycleManagerV1
I1125 04:08:30.079278       1 sync_worker.go:1036] Running sync for clusterrolebinding "cluster-olm-operator-role" (720 of 902)
I1125 04:08:30.079288       1 sync_worker.go:1039] Skipping clusterrolebinding "cluster-olm-operator-role" (720 of 902): disabled capabilities: OperatorLifecycleManagerV1
I1125 04:08:30.079295       1 sync_worker.go:1036] Running sync for deployment "openshift-cluster-olm-operator/cluster-olm-operator" (721 of 902)
I1125 04:08:30.079304       1 sync_worker.go:1039] Skipping deployment "openshift-cluster-olm-operator/cluster-olm-operator" (721 of 902): disabled capabilities: OperatorLifecycleManagerV1
I1125 04:08:30.079311       1 sync_worker.go:1036] Running sync for clusteroperator "olm" (722 of 902)
I1125 04:08:30.079320       1 sync_worker.go:1039] Skipping clusteroperator "olm" (722 of 902): disabled capabilities: OperatorLifecycleManagerV1
```

**Comment by jianzhangbjz** [comment]
> seems that olm operator creation is skipped. Is it expected?

No, why skip it? Any specific configures?

**Comment by evakhoni** [comment]
from the same log

```
I1125 04:08:10.571734       1 sync_worker.go:669] Previous sync status: &cvo.SyncWorkerStatus{Generation:3, Failure:error(nil), Done:0, Total:0, Completed:0, Reconciling:false, Initial:false, VersionHash:"", Architecture:"Multi", LastProgress:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), Actual:v1.Release{Architecture:"", Version:"4.18.0-0.nightly-multi-2024-11-22-231004", Image:"registry.build03.ci.openshift.org/ci-op-nym9y42l/release@sha256:e587424c309bde30865d8623949cd2d81fbb19e4aa07fb533143dfc2ba14d094", URL:"", Channels:[]string(nil)}, Verified:false, loadPayloadStatus:cvo.LoadPayloadStatus{Step:"PayloadLoaded", Message:"Payload loaded version=\"4.18.0-0.nightly-multi-2024-11-22-231004\" image=\"registry.build03.ci.openshift.org/ci-op-nym9y42l/release@sha256:e587424c309bde30865d8623949cd2d81fbb19e4aa07fb533143dfc2ba14d094\" architecture=\"Multi\"", AcceptedRisks:"", Failure:error(nil), Update:v1.Update{Architecture:"", Version:"4.18.0-0.nightly-multi-2024-11-22-231004", Image:"registry.build03.ci.openshift.org/ci-op-nym9y42l/release@sha256:e587424c309bde30865d8623949cd2d81fbb19e4aa07fb533143dfc2ba14d094", Force:false}, Verified:false, Local:true, LastTransitionTime:time.Time{wall:0xc1c91beaa20fe2c0, ext:370083311492, loc:(*time.Location)(0x48fffc0)}}, CapabilitiesStatus:cvo.CapabilityStatus{Status:v1.ClusterVersionCapabilitiesStatus{EnabledCapabilities:[]v1.ClusterVersionCapability{"CloudControllerManager", "CloudCredential", "Ingress", "MachineAPI"}, KnownCapabilities:[]v1.ClusterVersionCapability{"Build", "CSISnapshot", "CloudControllerManager", "CloudCredential", "Console", "DeploymentConfig", "ImageRegistry", "Ingress", "Insights", "MachineAPI", "NodeTuning", "OperatorLifecycleManager", "OperatorLifecycleManagerV1", "Storage", "baremetal", "marketplace", "openshift-samples"}}, ImplicitlyEnabledCaps:[]v1.ClusterVersionCapability(nil)}}
```
so OperatorLifecycleManagerV1 seems to be in EnabledCapabilities

**Comment by jianlinliu** [comment]
> from the same log
> 
> ```
> I1125 04:08:10.571734       1 sync_worker.go:669] Previous sync status: &cvo.SyncWorkerStatus{Generation:3, Failure:error(nil), Done:0, Total:0, Completed:0, Reconciling:false, Initial:false, VersionHash:"", Architecture:"Multi", LastProgress:time.Date(1, time.January, 1, 0, 0, 0, 0, time.UTC), Actual:v1.Release{Architecture:"", Version:"4.18.0-0.nightly-multi-2024-11-22-231004", Image:"registry.build03.ci.openshift.org/ci-op-nym9y42l/release@sha256:e587424c309bde30865d8623949cd2d81fbb19e4aa07fb533143dfc2ba14d094", URL:"", Channels:[]string(nil)}, Verified:false, loadPayloadStatus:cvo.LoadPayloadStatus{Step:"PayloadLoaded", Message:"Payload loaded version=\"4.18.0-0.nightly-multi-2024-11-22-231004\" image=\"registry.build03.ci.openshift.org/ci-op-nym9y42l/release@sha256:e587424c309bde30865d8623949cd2d81fbb19e4aa07fb533143dfc2ba14d094\" architecture=\"Multi\"", AcceptedRisks:"", Failure:error(nil), Update:v1.Update{Architecture:"", Version:"4.18.0-0.nightly-multi-2024-11-22-231004", Image:"registry.build03.ci.openshift.org/ci-op-nym9y42l/release@sha256:e587424c309bde30865d8623949cd2d81fbb19e4aa07fb533143dfc2ba14d094", Force:false}, Verified:false, Local:true, LastTransitionTime:time.Time{wall:0xc1c91beaa20fe2c0, ext:370083311492, loc:(*time.Location)(0x48fffc0)}}, CapabilitiesStatus:cvo.CapabilityStatus{Status:v1.ClusterVersionCapabilitiesStatus{EnabledCapabilities:[]v1.ClusterVersionCapability{"CloudControllerManager", "CloudCredential", "Ingress", "MachineAPI"}, KnownCapabilities:[]v1.ClusterVersionCapability{"Build", "CSISnapshot", "CloudControllerManager", "CloudCredential", "Console", "DeploymentConfig", "ImageRegistry", "Ingress", "Insights", "MachineAPI", "NodeTuning", "OperatorLifecycleManager", "OperatorLifecycleManagerV1", "Storage", "baremetal", "marketplace", "openshift-samples"}}, ImplicitlyEnabledCaps:[]v1.ClusterVersionCapability(nil)}}
> ```
> 
> so OperatorLifecycleManagerV1 seems to be in EnabledCapabilities looks like a bug to me 🐞

Agree, sounds like a bug in upgrade scenario.

**Comment by jianzhangbjz** [comment]
Hi @jianlinliu , can you share the product bug ID here? Thanks!

**Comment by jianlinliu** [comment]
> Hi @jianlinliu , can you share the product bug ID here? Thanks!

Hmm, we have no bug ID, we were proposing you to file a new bug.

**Comment by jinyunma** [comment]
@jianlinliu @jianzhangbjz Can we merge this PR first to unblock fresh installation failure? And upgrade issues can be tracked by new bug.

**Comment by jianzhangbjz** [comment]
> Hmm, we have no bug ID, we were proposing you to file a new bug.

The `OLMv1` was disabled as expected. See the finale clusterversion:https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/58914/rehearse-58914-periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-4.18-upgrade-from-stable-4.17-aws-ipi-sno-etcd-encryption-basecap-none-arm-f28/1860855242157985792/artifacts/aws-ipi-sno-etcd-encryption-basecap-none-arm-f28/gather-extra/artifacts/clusterversion.json
```console
            "spec": {
                "capabilities": {
                    "additionalEnabledCapabilities": [
                        "CloudCredential",
                        "CloudControllerManager",
                        "Ingress",
                        "MachineAPI"
                    ],
                    "baselineCapabilitySet": "None"
                },
                "channel": "candidate-4.17",
...
            "status": {
                "availableUpdates": null,
                "capabilities": {
                    "enabledCapabilities": [
                        "CloudControllerManager",
                        "CloudCredential",
                        "Ingress",
                        "MachineAPI"
                    ],
                    "knownCapabilities": [
                        "Build",
                        "CSISnapshot",
                        "CloudControllerManager",
                        "CloudCredential",
                        "Console",
                        "DeploymentConfig",
                        "ImageRegistry",
                        "Ingress",
                        "Insights",
                        "MachineAPI",
                        "NodeTuning",
                        "OperatorLifecycleManager",
                        "OperatorLifecycleManagerV1",
                        "Storage",
                        "baremetal",
                        "marketplace",
                        "openshift-samples"
                    ]
                },
```

**Comment by jianlinliu** [comment]
> that failure is an automation issue, not a product bug.

Per our past experience of handling newly introduced capacity operator. The script is doing right things. 

Per the [comment](https://github.com/openshift/release/blob/b0810632a5a2d73f18511b9f0b284085aa3483b6/ci-operator/step-registry/cucushift/upgrade/check/capability/cucushift-upgrade-check-capability-commands.sh#L375)
:
```
None baseline set should grow new caps of target version implicitly after upgrade
```

That means the new capacity operator in 4.18 should be automatically installed when 4.17 cluster upgrade completed, so that keep consistent like a fresh 4.18 install.

Per the test [log](https://gcsweb-qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/qe-private-deck/pr-logs/pull/openshift_release/58914/rehearse-58914-periodic-ci-openshift-openshift-tests-private-release-4.18-multi-nightly-4.18-upgrade-from-stable-4.17-aws-ipi-sno-etcd-encryption-basecap-none-arm-f28/1858789614408437760/artifacts/aws-ipi-sno-etcd-encryption-basecap-none-arm-f28/cucushift-upgrade-check-capability/build-log.txt) 

```
new_caps is OperatorLifecycleManagerV1
...
------check .status.conditions.ImplicitlyEnabledCapabilities------
actual_implicit_caps: 
expected_implicit_caps: OperatorLifecycleManagerV1
ERROR: Actual implicitly enabled capabilities don't match expected implicitly enabled capabilities
```

The new capacity operator - `OperatorLifecycleManagerV1` should be implicitly enabled once clusters get updated to 4.18, but it was not like that in this testing. 

I guess there is something missing in OperatorLifecycleManagerV1 operator manifest, unless it is by design for this specific operator.  @wking do you have some insight one this issue ?

**Comment by jianzhangbjz** [comment]
> That means the new capacity operator in 4.18 should be automatically installed when 4.17 cluster upgrade completed, so that keep consistent like a fresh 4.18 install.

Why? The `"baselineCapabilitySet"` is `"None"`, all optional capacities should NOT be installed unless you configure them manually.

**Comment by jianzhangbjz** [comment]
> Per our past experience of handling newly introduced capacity operator. The script is doing right things.

I guess they missed this scenario, OLMv1 wasn't installed on OCP 4.17, so why should it be installed as default on 4.18 after upgrading when `baseline cap is none`?  + @shellyyang1989 since she is this step owner based on  https://github.com/openshift/release/pull/45010

**Comment by jianlinliu** [comment]
> > Per our past experience of handling newly introduced capacity operator. The script is doing right things.
> 
> I guess they missed this scenario, OLMv1 wasn't installed on OCP 4.17, so why should it be installed as default on 4.18 after upgrading when `baseline cap is none`? + @shellyyang1989 since she is this step owner based on #45010

Okay, sounds reasonable, because OLMv1 is introduced in 4.18 for the 1st time, it is a bit different with other capacity operators. also cc @evakhoni and @jiajliu here, what do you think if the operator is not existing in 4.17, then upgrade to 4.18, the installed operators should be the same as 4.17 ?  If that is expected, then we need some update for the upgrade check capacity script.

**Comment by jiajliu** [comment]
> what do you think if the operator is not existing in 4.17, then upgrade to 4.18, the installed operators should be the same as 4.17 ? If that is expected, then we need some update for the upgrade check capacity script.

I just went through the design [doc](https://github.com/openshift/enhancements/blob/master/enhancements/installer/component-selection.md#updating-into-an-expanding-disabled-capability), it seems the new cap OLMv1(it is a first-introduced one) meet this scenario, so it might be disabled after upgrade.

```
Upgrade a cluster to a new version that includes new resources that belong to an addon that was excluded in the original install. The new resources should not be created.
```

**Comment by shellyyang1989** [comment]
>I guess they missed this scenario, OLMv1 wasn't installed on OCP 4.17, so why should it be installed as default on 4.18 after upgrading when baseline cap is none? + @shellyyang1989 since she is this step owner based on https://github.com/openshift/release/pull/45010I guess they missed this scenario, OLMv1 wasn't installed on OCP 4.17, so why should it be installed as default on 4.18 after upgrading when baseline cap is none? + @shellyyang1989 since she is this step owner based on https://github.com/openshift/release/pull/45010

Yeah, I didn't take the situation into account when I was working on it. Agreed with you that it looks like an automation issue.

**Comment by evakhoni** [comment]
strange one indeed. I guess its the first time ever that an optional capability wasn't a non-optional part of the cluster in a previous release. if that's the case, then yes, it is an exception. 

as stated above, any existing part of the cluster, be it a resource or an operator, regardless if its a capability or not, should never be disabled by becoming a capability. if such one is not explicitly enabled via baseline/additional, then it must be implicitlyEnabled by the cluster itself in order to not break that rule. 

with this one however, if I understand correctly, it does not exist in previous versions, and the first version it was introduced in, it was already an optional capability. do I understand it correctly?

**Comment by jianlinliu** [comment]
> with this one however, if I understand correctly, it does not exist in previous versions, and the first version it was introduced in, it was already an optional capability. do I understand it correctly?

Right.

**Comment by jianlinliu** [comment]
Per the above discussion, we are on the same page now, OLMv1 is an exception, we need to fix some automation script issue, I am going to fix it later.

/lgtm

**Pull request merged**